plugins/performance-lab/includes/site-health/audit-enqueued-assets/helper.php

```diff
@@ -279,31 +279,50 @@ function perflab_aea_get_total_size_bytes_enqueued_styles() {
 }
 
 /**
- * Convert full URL paths to absolute paths.
- * Covers Standard WP configuration, wp-content outside WP directories and subdirectories.
- * Ex: https://example.com/content/themes/, https://example.com/wp/wp-includes/
+ * Gets the size of the asset in bytes.
  *
- * @since 1.0.0
+ * @since n.e.x.t
  *
- * @param string $resource_url URl resource link.
- * @return string Returns absolute path to the resource.
+ * @param string $resource_url URL of the resource.
+ * @return int|null Returns the size in bytes of the asset, or null if it cannot be determined.
  */
-function perflab_aea_get_path_from_resource_url( string $resource_url ): string {
-	if ( '' === $resource_url ) {
-		return '';
-	}
+function perflab_aea_get_asset_size( string $resource_url ): ?int {
+	$response = wp_remote_get(
+		$resource_url,
+		array(
+			'timeout' => 10,
+			'headers' => perflab_aea_copy_basic_auth_headers( array() ),
+		)
+	);
 
-	// Different content folder ex. /content/.
-	if ( 0 === strpos( $resource_url, content_url() ) ) {
-		return WP_CONTENT_DIR . substr( $resource_url, strlen( content_url() ) );
+	if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+		return null;
 	}
 
-	// wp-content in a subdirectory. ex. /blog/wp-content/.
-	$site_url = untrailingslashit( site_url() );
-	if ( 0 === strpos( $resource_url, $site_url ) ) {
-		return untrailingslashit( ABSPATH ) . substr( $resource_url, strlen( $site_url ) );
+	$body = wp_remote_retrieve_body( $response );
+	if ( '' === $body ) {
+		return null;
 	}
 
-	// Standard wp-content configuration.
-	return untrailingslashit( ABSPATH ) . wp_make_link_relative( $resource_url );
+	return strlen( $body );
+}
+
+/**
+ * Copies HTTP Basic auth headers if present.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string, mixed> $headers Headers to copy to.
+ * @return array<string, mixed> Headers with copied Basic auth headers.
+ */
+function perflab_aea_copy_basic_auth_headers( array $headers ): array {
+	$basic_auth_header = sanitize_text_field( wp_unslash( $_SERVER['HTTP_AUTHORIZATION'] ?? '' ) );
+	if ( str_starts_with( $basic_auth_header, 'Basic ' ) ) {
+		$headers['Authorization'] = $basic_auth_header;
+	} elseif ( isset( $_SERVER['PHP_AUTH_USER'] ) && isset( $_SERVER['PHP_AUTH_PW'] ) ) {
```

Comment:
Is there a need for the `HTTP_AUTHORIZATION` here? I think [`wp_populate_basic_auth_from_authorization_header()`](https://github.com/WordPress/wordpress-develop/blob/972da8c86c7447c9a0571e020ec3e2ffdefdb00d/src/wp-includes/load.php#L97-L139) in core is ensuring that `$_SERVER['PHP_AUTH_USER']` and `$_SERVER['PHP_AUTH_PW']` are set 

Core is just using the [second condition](https://github.com/WordPress/wordpress-develop/blob/972da8c86c7447c9a0571e020ec3e2ffdefdb00d/src/wp-admin/includes/file.php#L545-L548) below, so this could be simplified as follows:

```suggestion
	if ( isset( $_SERVER['PHP_AUTH_USER'] ) && isset( $_SERVER['PHP_AUTH_PW'] ) ) {
```